components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -4540,6 +4580,254 @@ public void deleteApplicationByResourceId(String resourceId,
+                    while (resultSet.next()) {
+                        count = resultSet.getInt(1);
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            throw new IdentityApplicationManagementServerException("Error while getting discoverable application " +
+                    "basic information for resourceId: " + resourceId + " in tenantDomain: " + tenantDomain, e);
+        }
+        return count > 0;
+    }
+
+    private List<ApplicationBasicInfo> getDiscoverableApplicationBasicInfo(int limit, int offset, String
+            tenantDomain) throws IdentityApplicationManagementException {
+
+        List<ApplicationBasicInfo> applicationBasicInfoList = new ArrayList<>();
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            String databaseVendorType = connection.getMetaData().getDatabaseProductName();
+
+            try (NamedPreparedStatement statement =
+                         new NamedPreparedStatement(connection,
+                                 getDBVendorSpecificDiscoverableAppRetrievalQuery(databaseVendorType))) {
+                statement.setInt(ApplicationTableColumns.TENANT_ID, IdentityTenantUtil.getTenantId(tenantDomain));
+                statement.setInt(ApplicationConstants.OFFSET, offset);
+                statement.setInt(ApplicationConstants.LIMIT, limit);
+                statement.setInt(ApplicationConstants.ZERO_BASED_START_INDEX, offset);
+                statement.setInt(ApplicationConstants.ONE_BASED_START_INDEX, offset + 1);
+                statement.setInt(ApplicationConstants.END_INDEX, offset + limit);
+
+                try (ResultSet resultSet = statement.executeQuery()) {
+                    while (resultSet.next()) {
+                        applicationBasicInfoList.add(buildApplicationBasicInfo(resultSet));
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            throw new IdentityApplicationManagementServerException("Error while getting application basic information" +
+                    " for discoverable applications in tenantDomain: " + tenantDomain, e);
+        }
+
+        return Collections.unmodifiableList(applicationBasicInfoList);
+    }
+
+    private int getCountOfDiscoverableApplications(String tenantDomain) throws IdentityApplicationManagementException {
+
+        int count;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+
+            try (NamedPreparedStatement statement =
+                         new NamedPreparedStatement(connection, LOAD_DISCOVERABLE_APP_COUNT_BY_TENANT)) {
+                statement.setInt(ApplicationTableColumns.TENANT_ID, IdentityTenantUtil.getTenantId(tenantDomain));
+
+                try (ResultSet resultSet = statement.executeQuery()) {
+                    resultSet.next();
+                    count = resultSet.getInt(1);
```

Comment:
can directly return from here.

Reply:
I think the way it's written is more readable